app/src/main/kotlin/github/com/rhacco/dota2androidapp/activities/LiveMatchesActivity.kt

```diff
@@ -29,21 +39,26 @@ class LiveMatchesActivity : BaseLifecycleActivity<MatchesViewModel>() {
     }
 
     // Add tournament matches first, then sort by average MMR (descending)
-    private fun updateLiveMatchesListData(listItemData: LiveMatchesListAdapter.ListItemData) {
-        if (listItemData.mAverageMMR < 1) {
-            mListAdapter.mListItemsData.add(0, listItemData)
+    private fun updateLiveMatchesListData(newItem: LiveMatchesListAdapter.ListItemData) {
+        // Strangely, sometimes the same match arrives here twice, so prevent creating duplicates
+        mListAdapter.mListItemsData
+                .filter { it.mMatchID == newItem.mMatchID }
+                .forEach { return }
+
+        if (newItem.mAverageMMR < 1) {
+            mListAdapter.mListItemsData.add(0, newItem)
             return
         }
         var index = 0
         for (item in mListAdapter.mListItemsData) {
-            if (item.mAverageMMR < 1 || item.mAverageMMR >= listItemData.mAverageMMR) {
+            if (item.mAverageMMR < 1 || item.mAverageMMR >= newItem.mAverageMMR) {
```

Comment:
What is the idea behind  || item.mAverageMMR >= newItem.mAverageMMR?

Reply:
Sorting by descending average MMR (see comment for method)